### PR TITLE
Refactor and add docs docs docs

### PR DESCRIPTION
**Pull request by FiloSottile**

### CODE REVIEW

File: internal/radix51/fe.go [hunk 1]
```diff
@@ -15,13 +18,12 @@ import (
 
 // FieldElement represents an element of the field GF(2^255-19). An element t
 // represents the integer t[0] + t[1]*2^51 + t[2]*2^102 + t[3]*2^153 +
-// t[4]*2^204. The zero value is a valid zero element.
+// t[4]*2^204. Each limb must not exceed 2^54 - 1 to be valid.
```
Comment (gtank) [inline internal/radix51/fe.go]: Limbs are allowed to grow between operations, but must not exceed 54 bits to remain valid input for multiplications.

File: internal/radix51/fe.go [hunk 2]
```diff
@@ -263,10 +263,10 @@ func (v *FieldElement) FromBytes(x []byte) *FieldElement {
 	return v
 }
 
-func (v *FieldElement) ToBytes(r []byte) {
-	if len(r) != 32 {
-		panic("invalid input size")
-	}
+// AppendBytes appends a 32 bytes little-endian encoding of v to b.
+func (v *FieldElement) AppendBytes(b []byte) []byte {
```
[the remark this answers is not in the document]
Comment (FiloSottile) [reply internal/radix51/fe.go]: In general it always feels weird to pass fixed sized slices of zeroes in Go.

But yeah, I'm not positive about this either.

File: internal/radix51/fe.go (hunk 1 above)
Comment (FiloSottile) [reply internal/radix51/fe.go]: Ok, I shouldn't write docs about this without knowing what I'm doing.

We really need to figure out the input and output ranges for each operation, and either make it all self-consistent, or if that's not possible put the ranges in the docs.

File: internal/radix51/fe.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (FiloSottile) [reply internal/radix51/fe.go]: Renamed Bytes.

**Pull request merged**